src/components/tracked-controls.js

```diff
@@ -35,6 +36,27 @@ module.exports.Component = registerComponent('tracked-controls', {
 
     this.previousControllerPosition = new THREE.Vector3();
     this.updateGamepad();
+    this.onComponentInitialized = bind(this.onComponentInitialized, this);
+  },
+
+  play: function () {
+    this.el.addEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  pause: function () {
+    this.el.removeEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  onComponentInitialized: function (evt) {
+    if (evt.detail.name !== 'tracked-controls') { return; }
+    var self = this;
+    Object.keys(this.el.components).forEach(function (name) {
+      const component = self.el.components[name];
+      if (component.controllerPresent) {
+        // Emit controllerpresent event.
+        self.el.emit('controllerpresent', {name: name, component: component});
```

Comment:
so the a-blast use case is kinda like `hand-controls` usage -- I think it's called `shoot-controls` -- but it needs to specifically know which higher-level controller component (`oculus-touch-controls` and `vive-controls`) to adjust model pose, etc.

I don't follow how `oculus-touch-controls` and `vive-controls` doing the subscription helps `shoot-controls`, and that subscription wouldn't help `shoot-controls` since it only knows `oculus-touch-controls` and `vive-controls`, it doesn't know the details of how they recognize what controller/gamepad.